twill-yarn/src/main/java/org/apache/twill/yarn/YarnTwillPreparer.java

```diff
@@ -455,21 +475,40 @@ private void saveLogLevels(String runnableName, Map<String, LogEntry.Level> logL
     this.logLevels.put(runnableName, newLevels);
   }
 
+  /**
+   * Creates an {@link Credentials} by copying the {@link Credentials} of the current user.
+   */
   private Credentials createCredentials() {
     Credentials credentials = new Credentials();
 
     try {
       credentials.addAll(UserGroupInformation.getCurrentUser().getCredentials());
+    } catch (IOException e) {
```

Comment:
I guess app submission would fail if caller doesn't acquire required credentials in its own way anyways.